### PR TITLE
frontend: type 2 subscription fns signProgresss and signConfirm 

### PR DESCRIPTION
**Pull request by shonsirsha**
type 2 subscription fns `signProgresss` and `signConfirm`

On top of that did a small refactor creating a reusable `updateBalance` fn.

### CODE REVIEW

File: frontends/web/src/routes/account/send/send.tsx [hunk 1]
```diff
@@ -158,28 +158,10 @@ class Send extends Component<Props, State> {
...
-              this.setState({ signProgress: meta, signConfirm: false });
-              break;
-            case 'signConfirm':
-              this.setState({ signConfirm: true });
-              break;
-            }
-            break;
-          }
-        }
-      }),
```
Comment (thisconnect) [inline frontends/web/src/routes/account/send/send.tsx:177]: Can you type these 2 subscriptions in api/devicessync?

It should be similar to channelHashChanged in api/devicessync.ts

**Review by thisconnect — CHANGES_REQUESTED** [review]

**Comment by shonsirsha** [comment]
Made the change, PTAL :) updaated commit msg and rebased as well.

File: frontends/web/src/api/devicessync.ts [hunk 2]
```diff
@@ -77,3 +79,24 @@ export const attestationCheckDone = (
   });
   return unsubscribe;
 };
+
+
+export const signProgress = (
+  cb: (meta: TSignProgress) => void
+): TUnsubscribe => {
+  return apiWebsocket(payload => {
```
Comment (thisconnect) [inline frontends/web/src/api/devicessync.ts:87]: just use subscribeLegacy like the other functions in this module do.

have a look at frontends/web/src/utils/event-legacy.ts there it uses apiWebsocket internally

File: frontends/web/src/api/devicessync.ts [hunk 3]
```diff
@@ -77,3 +79,24 @@ export const attestationCheckDone = (
...
+    if ('type' in payload && payload.type === 'device' && payload.data === 'signProgress') {
+      cb(payload.meta);
+    }
+  });
+};
+
+export const signConfirm = (
+  cb: (signConfirm: boolean) => void
+): TUnsubscribe => {
+  return apiWebsocket(payload => {
```
Comment (thisconnect) [inline frontends/web/src/api/devicessync.ts:97]: same, change to subscribeLegacy

File: frontends/web/src/api/devicessync.ts [hunk 4]
```diff
@@ -77,3 +79,24 @@ export const attestationCheckDone = (
...
+    if ('type' in payload && payload.type === 'device' && payload.data === 'signProgress') {
+      cb(payload.meta);
+    }
+  });
+};
+
+export const signConfirm = (
+  cb: (signConfirm: boolean) => void
```
Comment (thisconnect) [inline frontends/web/src/api/devicessync.ts:95]: I think the backend doesn't return anything here, but maybe check if there is something in the event object, or check this handler in the backend go implementation.

so I thin you can just use `() => void` to use this as a callback.

and then change

```js
  signConfirm(() =>
    this.setState({ signConfirm: true })
  ),
```

File: frontends/web/src/routes/account/send/services.ts [hunk 5]
```diff
@@ -18,6 +18,11 @@
 import { alertUser } from '../../../components/alert/Alert';
 import { i18n } from '../../../i18n/i18n';
 
+export type TSignProgress = {
+  steps: number;
+  step: number;
+}
```
Comment (thisconnect) [inline frontends/web/src/routes/account/send/services.ts:24]: this type is coming from the backend api and should be in api/devicessync.ts

**Review by thisconnect — CHANGES_REQUESTED** [review]

File: frontends/web/src/routes/account/send/services.ts (hunk 5 above)
Comment (shonsirsha) [reply frontends/web/src/routes/account/send/services.ts:24]: Makes sense 👍

File: frontends/web/src/routes/account/send/send.tsx [hunk 6]
```diff
@@ -158,28 +158,15 @@ class Send extends Component<Props, State> {
...
+      signProgress((meta) =>
```
Comment (thisconnect) [inline frontends/web/src/routes/account/send/send.tsx:161]: nit: meta is event internal.. Maybe name it progress or similar?

```suggestion
      signProgress((progress) =>
```

**Review by thisconnect — APPROVED** [review]
untested LGTM with a small nit

**Pull request merged**